src/rebar_prv_compile.erl

```diff
@@ -195,13 +195,14 @@ build_app(AppInfo, State) ->
     case rebar_app_info:project_type(AppInfo) of
         Type when Type =:= rebar3 ; Type =:= undefined ->
             Compilers = rebar_state:compilers(State),
+            rebar_paths:set_paths([deps], State),
             rebar_compiler:compile_all(Compilers, AppInfo);
         Type ->
             ProjectBuilders = rebar_state:project_builders(State),
             case lists:keyfind(Type, 1, ProjectBuilders) of
                 {_, Module} ->
                     %% load plugins since thats where project builders would be
-                    rebar_paths:set_paths([plugins, deps], State),
+                    rebar_paths:set_paths([deps, plugins], State),
```

Comment:
The change at line 198 looks right, but is this really needed?

Hm, not even sure we should even just be leaving it up to the `Module:build` function to handle if it needs deps loaded, we just need plugins since the Module is in a plugin.

Reply:
yeah my way of thinking is that it's unlikely that a module needs itself to build as a builder, therefore we should be safe loading the plugins _after_ the deps, so that at least the compile-time stuff uses the right app-level dependencies during its build

Reply:
Yea, just not sure we need deps at all. Like the mix builder certainly doesn't. But it shouldn't hurt anything so may as well keep it.

Reply:
yeah I'd only expect it if a custom compiler ends up needing something like parse transforms or the .hrl files' path importance. Currently probably no one uses it.